modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigDropDown.kt

```diff
@@ -1,197 +1,138 @@
 
-    // mostly what is viewed by the user is done here, goal is to have minimal logic in the function
-    // because it is called every half a second
-    override fun update(e: AnActionEvent) {
-        e.project?.let { project ->
-            // this check ensures that we don't query the index when it is being built
-            // querying the index during the startup/Dumb Mode can give us 0 or stale data
-            if (!::configFiles.isInitialized) {
-                e.presentation.isVisible = false
-                blockAndQueryIndex(project)
-                return
+            val configs: MutableMap<String, () -> Unit> = mutableMapOf()
+            service.activeConfig?.let {
+                configs["(active) %s".format(it.presentableUrl)] = {
+                    FileEditorManager.getInstance(service.project).openFile(it, true)
+                }
             }
 
-            blockAndQueryIndex(project)
-
-            if (configFiles.size > 1) {
-                if (chosenFile !in configFiles) {
-                    chosenFile = configFiles.first()
+            val defaultConfig = service.configApi.getDefaultConfig()
+            if (defaultConfig == null) {
+                configs["(create default)"] = {
+                    val config = service.configApi.createDefaultConfig()
+                    FileEditorManager.getInstance(service.project).openFile(config, true)
                 }
-
-                e.presentation.text = chosenFile?.let { getReadablePath(it, project) }
             } else {
-                chosenFile = configFiles.firstOrNull()
+                configs["(default) %s".format(defaultConfig.presentableUrl)] = {
+                    FileEditorManager.getInstance(service.project).openFile(defaultConfig, true)
+                }
             }
-            e.presentation.isVisible = configFiles.size > 1
-        }
-    }
 
-    private fun updateConfigFiles(project: Project) {
-        val updatedConfigFiles = HashSet<String>()
-        val basePath = project.basePath ?: throw Error("couldn't resolve project path")
-        val allKeys = FileBasedIndex.getInstance().getAllKeys(MirrordConfigIndex.key, project)
-        // to get the updated keys, we need to use this particular method. no other methods such getAllKeys, processAllKeys
-        // give updated values
-        FileBasedIndex.getInstance().processFilesContainingAnyKey(
-            MirrordConfigIndex.key,
-            allKeys,
-            GlobalSearchScope.projectScope(project),
-            null,
-            null
-        ) {
-            if (it.path.startsWith(basePath)) {
-                updatedConfigFiles.add(it.path)
+            val options = configs.keys.toList().sorted()
+
+            val selected = if (options.size == 1) {
+                options.first()
+            } else if (options.size > 1) {
+                MirrordConfigDialog("Edit mirrord configuration", options).show()?.option
+            } else {
+                null
```

Comment:
kotlin suggestion, i think we can simplify this:

```kt
val options = configs.keys.toList().sorted()

val selected = options.singleOrNull()?.let { singleOption ->
    MirrordConfigDialog("Edit mirrord configuration", options).show()?.option ?: singleOption
}
```

Reply:
This actually doesn't do what we want, but I changed the expression to be more "chainy"